frontend/src/components/custom-tools/output-analyzer-card/OutputAnalyzerCard.jsx

```diff
@@ -11,26 +11,38 @@ import { useSessionStore } from "../../../store/session-store";
 import { CombinedOutput } from "../combined-output/CombinedOutput";
 import { DocumentViewer } from "../document-viewer/DocumentViewer";
 import { PdfViewer } from "../pdf-viewer/PdfViewer";
+import { useParams } from "react-router-dom";
 
+let publicDocumentApi;
+try {
+  publicDocumentApi =
+    require("../../../plugins/prompt-studio-public-share/helpers/PublicShareAPIs").publicDocumentApi;
+} catch {
+  // The component will remain null of it is not available
+}
 function OutputAnalyzerCard({ doc, totalFields }) {
   const [fileUrl, setFileUrl] = useState("");
   const [isDocLoading, setIsDocLoading] = useState(false);
   const [filledFields, setFilledFields] = useState(0);
-  const { details } = useCustomToolStore();
+  const { details, isPublicSource } = useCustomToolStore();
   const { sessionDetails } = useSessionStore();
   const { setAlertDetails } = useAlertStore();
   const axiosPrivate = useAxiosPrivate();
   const handleException = useExceptionHandler();
+  const { id } = useParams();
```

Comment:
@harini-venkataraman You can get the tool ID from `details` (inside the custom tool store). Can you please replace its usage with `details?.tool_id` here and wherever applicable? This way, we can avoid an unnecessary import.

Reply:
@tahierhussain ID here refers to the `share_id` for public prompt studio. Is it okay to fetch it from params here?

Reply:
Sure, @harini-venkataraman. If we're using the tool_id anywhere from the params, it can be taken from the `details` object. Resolving the comment.